extensions/ql-vscode/src/compare-performance/compare-performance-view.ts

```diff
@@ -12,16 +21,27 @@ import { showAndLogExceptionWithTelemetry } from "../common/logging";
 import { extLogger } from "../common/logging/vscode";
 import type { WebviewPanelConfig } from "../common/vscode/abstract-webview";
 import { AbstractWebview } from "../common/vscode/abstract-webview";
+import type { ProgressCallback } from "../common/vscode/progress";
+import { reportStreamProgress, withProgress } from "../common/vscode/progress";
 import { telemetryListener } from "../common/vscode/telemetry";
-import type { HistoryItemLabelProvider } from "../query-history/history-item-label-provider";
-import { PerformanceOverviewScanner } from "../log-insights/performance-comparison";
-import { scanLog } from "../log-insights/log-scanner";
+import { downloadTimeout } from "../config";
 import type { ResultsView } from "../local-queries";
+import { scanLog } from "../log-insights/log-scanner";
+import { PerformanceOverviewScanner } from "../log-insights/performance-comparison";
+import type { HistoryItemLabelProvider } from "../query-history/history-item-label-provider";
+import { tmpDir } from "../tmp-dir";
+
+type ComparePerformanceCommands = {
+  "codeQL.compare-performance.downloadExternalLogs": () => Promise<void>;
+};
 
 export class ComparePerformanceView extends AbstractWebview<
   ToComparePerformanceViewMessage,
   FromComparePerformanceViewMessage
 > {
+  private workingDirectory;
```

Comment:
```suggestion
  private workingDirectory: string;
```
Does TypeScript infer types of fields with an initializer now, or does this have type `any`?

Reply:
it is typed as `string` everywhere 🎉